reactComponents/src/searchbar/tokenSearch/SearchInput.tsx

```diff
@@ -1,90 +1,97 @@
+    border-radius: ${ props?.styles?.borderRadius || "0" };  
+    background: ${ props?.styles?.background || "#08333c" };   
+    padding: ${ props?.styles?.padding || "11px 15px" };    
+    font-size: ${ props?.styles?.fontSize || "15px" };      
+    font-family: ${ props?.styles?.fontFamily || "'Fira Code', monospace" };
+  `}  
 `;
 
-const StyledInput = styled.input`
-  width: 100%;
-  border: none;
-  background-color: inherit;
-  color: #ffffff;
-  //width: auto;
-`;
-
-const HideOnSmallScreen = styled.img`
-  width: 30px;
+const StyledSearchIconWrapper = styled.div`  
   cursor: pointer;
   float: right;
   position: absolute;
-  right: 22px;
-  top: 9px;
-  @media only screen and (max-width: 990px) {
-    display: none;
-  }
+  ${({props}) => `
+    right: ${ props?.styles?.right || "15px" };      
+    top: ${ props?.styles?.top || "12px" };        
+  `}    
 `;
 
+const StyledWrapper = styled.div`
+  position: relative;
+`;
 
 const SearchInput = () => {
   const dispatch = useDispatch();
+  const renderProps = useContext(TokenSearchContext);  
+  const { customSearchInput } = renderProps;
+ 
   const { searchText, networkMap, exchangeMap } = useSelector((state:RootState) => state);
   
-  // Updates the datasets of the results.
+  // Updates the datasets of the results. 
   useEffect(() => {
     // Ensure that the search text fulfills the minimum lenght requirement.
     if (searchText.length >= minStringSearch) {
       dispatch(searchTokenPairs(searchText));
     }
-  }, [dispatch, searchText, networkMap, exchangeMap]);
-
+  }, [dispatch, searchText, networkMap, exchangeMap]); 
+  
 
   const onChangeFilter = (event) => {    
     const value = event.target.value    
     dispatch(setSearchText(value))
   }
 
-  const debounceChangeHandler = useCallback( debounce(onChangeFilter, 350), [searchText])  
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debounceChangeHandler = useCallback( debounce(onChangeFilter, 350), [searchText])
+  
+  const placeholder = customSearchInput?.placeholder ?  customSearchInput?.placeholder : 'Please input token name or address.'
+  const activeColor = customSearchInput?.styles?.search?.activeColor ? customSearchInput?.styles?.search?.activeColor : '#666699'
+  const color  = customSearchInput?.styles?.search?.color ? customSearchInput?.styles?.search?.color : '#FFF'
+  const height = customSearchInput?.styles?.search?.height ? customSearchInput?.styles?.search?.height : 14
+  const width = customSearchInput?.styles?.search?.width ? customSearchInput?.styles?.search?.width : 14
+
   // RENDERING.
   return (
-    <PairField onClick={() => dispatch(startSelecting())}>
-      <StyledInput
-        placeholder={'Select a token pair'}
+    <StyledWrapper onClick={() => dispatch(startSelecting())}>
+      <StyledInput 
+        placeholder={placeholder}
         autocomplete={'off'}
         onChange={debounceChangeHandler}
+        styles={customSearchInput?.styles?.input}
```

Comment:
no default value here, right?

Reply:
Yes

Reply:
If you see the StyledXXX wrap, it has the defaultvalue.